src/main/java/org/openrewrite/maven/AbstractRewriteBaseRunMojo.java

```diff
@@ -148,6 +149,14 @@ protected ResultsContainer listResults(ExecutionContext ctx) throws MojoExecutio
                 }
             }
 
+            getLog().info("Check recipe licences...");
+            failedValidations = checkLicence(recipe);
```

Comment:
Right now we're validating the active recipe and any recipes that contains. I had expected we'd already throw earlier when loading recipes at all. Now folks can still misconfigure and only find out depending on what they run. Do you see an option for that?

Reply:
We could validate at classloading, but for the Maven plugin thats only seconds, or less, before validating the active recipe like now. 
Maybe we could hook into the resolution of the dependencies and break when a artifact is downloaded that is Moderne Prop. but again seconds before the current point.
So what advantage do you see and which goal you have in mind?